test/integration/auth-methods/jwt.test.js

```diff
@@ -0,0 +1,1664 @@
+        ...ESAlgorithmsWithPassPhrase
+        // verify whether RSASSA-PSS (PS) algorithm is applicable
+    },
+
+    algorithms = Object.entries(algorithmsSupported);
+
+describe('jwt auth', function () {
+    let testrun;
+
+    // with invalid algorithm - root level
+    algorithms.forEach(([key]) => {
+        const { alg, signKey } = algorithmsSupported[key];
+
+        describe(`with invalid algorithm for ${alg} algorithm`, function () {
+            before(function (done) {
+                const runOptions = {
+                    collection: {
+                        item: {
+                            request: {
+                                url: 'https://postman-echo.com/headers',
+                                auth: {
+                                    type: 'jwt',
+                                    jwt: {
+                                        algorithm: '', // invalid root level algo
+                                        header: { alg },
+                                        payload: { test: 123 },
+                                        secret: signKey,
+                                        addTokenTo: HEADER
+                                    }
+                                }
+                            }
+                        }
+                    }
+                };
+
+                this.run(runOptions, function (err, results) {
+                    testrun = results;
+                    done(err);
+                });
+            });
+
+            it('should completed the run', function () {
+                expect(testrun).to.be.ok;
+                expect(testrun).to.nested.include({
+                    'done.calledOnce': true,
+                    'start.calledOnce': true,
+                    'request.calledOnce': true
+                });
+            });
+
+            it('should not add Authorization header', function () {
+                const headers = [],
+                    request = testrun.request.firstCall.args[3];
+
+                request.headers.members.forEach(function (header) {
+                    headers.push(header.key);
+                });
+
+                expect(headers).that.does.not.include('Authorization');
+            });
+        });
+    });
+
+    // with invalid header
+    algorithms.forEach(([key]) => {
+        const { alg, signKey } = algorithmsSupported[key];
+
+        describe(`with invalid header for ${alg} algorithm`, function () {
+            before(function (done) {
+                const runOptions = {
+                    collection: {
+                        item: {
+                            request: {
+                                url: 'https://postman-echo.com/headers',
+                                auth: {
+                                    type: 'jwt',
+                                    jwt: {
+                                        algorithm: alg,
+                                        header: null,
```

Comment:
We should also test the case where `header` is an object with extra keys but invalid `alg` key.
```suggestion
                                        header: {
                                        	alg: 'some other alg',
                                        	myKey: 'myVal'
                                        },
```